Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -432,10 +432,10 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::locale_compare)
     auto that_value = TRY(vm.argument(0).to_string(vm));
 
     // 4. Let collator be ? Construct(%Collator%, « locales, options »).
-    auto* collator = TRY(construct(vm, *realm.intrinsics().intl_collator_constructor(), vm.argument(1), vm.argument(2)));
+    auto collator = TRY(construct(vm, *realm.intrinsics().intl_collator_constructor(), vm.argument(1), vm.argument(2)));
 
     // 5. Return CompareStrings(collator, S, thatValue).
-    return Intl::compare_strings(static_cast<Intl::Collator&>(*collator), Utf8View(string), Utf8View(that_value));
+    return Intl::compare_strings(static_cast<Intl::Collator&>(*collator.ptr()), Utf8View(string), Utf8View(that_value));
```

Comment:
Wouldn't just `*collator` give you a reference here already?

Reply:
yep, fixed